### ISSUE TITLE
Manually enter the routing path in the browser address bar and report an error 404

### ISSUE THREAD

**Issue by zhenzhenChange** [vuejs/vue-router-next#470]
### Version
4.0.0-beta.10

### Reproduction link
[https://github.com/zhenzhenChange/awaking](https://github.com/zhenzhenChange/awaking)

### Steps to reproduce
I directly typed a `path` in `routes` in the browser address bar, the console reported an error 404, but the `to` prop of `router-link` can be used to jump correctly.But after the jump is completed, F5 refreshes the page and also reports a 404 error.Is this a bug? Or is there no such feature yet? Or set it on purpose?

### What is expected?
I expect to enter the route path in the address bar to jump to the corresponding view

### What is actually happening?
In fact, there is no jump, the console reports an error 404, and the page displays Not Found

---
If there is a corresponding document or issue description, please let me know, thank you
- git clone https://github.com/zhenzhenChange/awaking
- cd doraemon
- yarn
- yarn dev
- open browser
- Use the `router-link` in the lower right corner of the picture to jump (success)
- Enter `path` directly in the address bar or F5 refresh will report an error 404

Trouble: I can actually test in the [codesandbox](https://codesandbox.io/s/vue-router-v4-reproduction-tk1y7?file=/index.html) provided by you.

**Comment by posva** [comment]
you didn't add a catch all route https://next.router.vuejs.org/guide/essentials/dynamic-matching.html#catch-all-404-not-found-route

**Comment by zhenzhenChange** [comment]
Very sorry for interrupting you.
First of all thank you for your reply. Then I found the reason now.
This is a problem with the location of `index.html`. In the initialization template of `Vite`, the directory structure is like this:
- vite-demo
  - public
  - src
  - index.html （Same level as `public` folder）

But due to past development habits, I moved it to the public folder：
- vite-demo
  - public
    - index.html
  - src

This leads to a problem with the path referenced by the `src` prop of the `script` tag in the `index.html` file：
```js
<script type="module" src="/src/main.ts"></script>
```
I didn't pay attention to this problem

I found out by comparing a new vite project. And pressing F5 to refresh the browser causes the 404 problem, which is similar to the problem of the vue-router conflicting with the URL when the vue project is deployed on the server (this usually requires additional configuration on the server). Of course, in the vite project, it may be able to pass vite .confit.js file configuration, I am still studying...
